src/types.ts

```diff
@@ -104,9 +114,11 @@ export type IWorkspacesSection = Array<string>
 
 /**
  * Yarn is a special snowflake.
+ * So is PNPM.
  */
-export interface IYarnWorkspacesConfig {
-  packages: IWorkspacesSection
+export interface IYarnPnpmWorkspacesConfig {
```

Comment:
I think it might be time to come up with a proper abstraction. I even forgot why Yarn was a special snowflake. 🙃

Reply:
Yarn allows turning off hoisting packages, so _sometimes_ it's different from npm. Conteidentally, it's a superset of the format pnpm uses, except that pnpm uses yaml and a different file.